### ISSUE TITLE
Wrong behaviour of sending back to address when using TREZOR

### ISSUE THREAD

**Issue by prusnak** [spesmilo/electrum#3920]
If one uses TREZOR and unticks "Use change address" in Preferences > Transactions, the transaction being signed by TREZOR does not indicate that "send-back-to-me" output is, in fact, a change address. This can be easily fixed by using `address_n` instead of `address` field in `TxOutputType` structure.

**Comment by SomberNight** [comment]
Can there still be only one change address? It would be easiest to just use `address_n` for all outputs controlled by the wallet and not special case this.

**Comment by SomberNight** [comment]
Well I am not sure what I am doing wrong but I can't get this to work by simply setting `address_n`.

-----

1-in 2-out txn where the "input address" is a receiving address -- the trezor prompts me to confirm both outputs
```
trezor.tx_outputs() [<TxOutputType: {'address': 'tb1qlkl7d8q2t5tzgypa275hfq0l0dm850xg7pztf9', 'amount': 5000, 'script_type': 0}>, <TxOutputType: {'amount': 994859, 'address_n': [2147483732, 2147483649, 2147483648, 0, 5], 'script_type': 4}>]
```

-----

1-in 2-out txn where the "input address" is a change address -- the trezor only prompts me to confirm the non-change output
```
trezor.tx_outputs() [<TxOutputType: {'address': 'tb1qlkl7d8q2t5tzgypa275hfq0l0dm850xg7pztf9', 'amount': 5000, 'script_type': 0}>, <TxOutputType: {'amount': 10059, 'address_n': [2147483732, 2147483649, 2147483648, 1, 6], 'script_type': 4}>]

```

python-trezor 0.9.0
trezor fw 1.6.0

**Comment by SomberNight** [comment]
TBH, I am completely unfamiliar with trezor fw code, but based on these
- https://github.com/trezor/trezor-mcu/blob/e1fa7af1da79e86ccaae5f3cd2a6c4644f546f8a/firmware/signing.c#L551
- https://github.com/trezor/trezor-mcu/blob/v1.6.0/firmware/signing.c#L394
- https://github.com/trezor/trezor-mcu/blob/e1fa7af1da79e86ccaae5f3cd2a6c4644f546f8a/firmware/signing.c#L579
- https://github.com/trezor/trezor-mcu/blob/e1fa7af1da79e86ccaae5f3cd2a6c4644f546f8a/firmware/transaction.c#L252-L254

it looks to me that unless an output is specifically on the `.../1/*` bip32 subtree, the device will prompt me to confirm it, hence it looks what you are asking for is not possible on our end?

**Comment by prusnak** [comment]
Right. Have not realized, we check whether the change address comes from the `x/1/*` subtree, where `x/0` is the path that needs to be shared among all inputs. Maybe we should also accept `x/0` as a valid change address path. @jhoenicke what do you think?

**Comment by jhoenicke** [comment]
I'm not opposed to allowing /0/* in addition to /1/*. It would also be useful for colored coins, e.g., the omni protocol where the change address must be equal to the send address.

We currently only allow one output to a change address.   We should change this behavior to allow multiple outputs, but treat only the first as a hidden change output and ask for confirmations for the others.  And I'd like to add an indication for non-standard change outputs that they are controlled by the Trezor, e.g. show the account#/path below the address like in the display address dialog.

**Comment by kyuupichan** [comment]
@jhoenicke agreed, the current behaviour is sub-optimal for privacy

**Comment by prusnak** [comment]
> @jhoenicke agreed, the current behaviour is sub-optimal for privacy

How exactly?

**Comment by kyuupichan** [comment]
As I explained 2 years ago; limited options for change outputs is atrocious for privacy

**Comment by prusnak** [comment]
I merged the PR by Jochen, so next version of TREZOR firmware will deploy the fix implemented here: https://github.com/trezor/trezor-mcu/pull/306

**Comment by prusnak** [comment]
Fixed deployed yesterday in TREZOR firmware 1.6.1
